src/funcx_common/task_storage/redis.py

```diff
@@ -0,0 +1,60 @@
+import typing as t
+
+from ..tasks import TaskProtocol
+from .base import StorageException, TaskStorage
+
+
+class RedisTaskStorage(TaskStorage):
+    """
+    RedisImplicitTaskStorage stores values in redis by returning the
+    target string in a wrapper so that the storage request is reverse
+    delegated to the caller which stores the task.<fields> into REDIS
+
+    This relies on the caller having a mechanism in place to store
+    values into redis
+    """
+
+    storage_id = "Redis"
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def store_result(self, task: TaskProtocol, result: str) -> bool:
+        task.result = result
+        task.result_reference = {"storage_id": self.storage_id}
+        return True
+
+    def get_result(self, task: TaskProtocol) -> t.Optional[str]:
+        if (
+            task.result_reference
+            and task.result_reference["storage_id"] == self.storage_id
+        ):
+            return task.result
+        else:
+            raise StorageException(f"Task not stored with {self.storage_id}")
+
+
+class ThresholdedRedisTaskStorage(RedisTaskStorage):
+    """
+    RedisImplicitTaskStorage stores values in redis by returning the
+    target string in a wrapper so that the storage request is reverse
+    delegated to the caller which stores the task.<fields> into REDIS
+
+    This relies on the caller having a mechanism in place to store
+    values into redis
+    """
+
+    storage_id = "ThresholdedRedis"
+
+    def __init__(self, result_limit_chars: int = 100) -> None:
+        super().__init__()
+        self.result_limit_chars = result_limit_chars
+
+    def store_result(self, task: TaskProtocol, result: str) -> bool:
+        if len(result) <= self.result_limit_chars:
+            task.result = result
+            return True
```

Comment:
This is not setting `task.result_reference`. You could just use `super().store_result` in this case.

The fact that this is missing tells me that there's no test running on this functionality.
Since the storage mechanism in these classes doesn't operate directly on Redis, but delegates to the Task object, you could presumably write a few tests for this using the SimpleInMemoryTask object defined in the testsuite. You just need to make sure that the object gets `result` and `result_reference` set correctly.

Reply:
Outdated.

Reply:
This is marked as resolved, but unless I'm mistaken, `ThresholdedRedisTaskStorage.store_result` still does not set `task.result_reference`.